src/Nethermind/Nethermind.Blockchain/BlockTree.AcceptVisitor.cs

```diff
@@ -77,6 +77,12 @@ public async Task Accept(IBlockTreeVisitor visitor, CancellationToken cancellati
                         }
                         else
                         {
+                            if (block.Header.TotalDifficulty is null || block.Header.TotalDifficulty == 0)
+                            {
+                                if (_logger.IsTrace) _logger.Trace($"Setting TD for block {block.Number}. Old TD: {block.TotalDifficulty}.");
+                                SetTotalDifficulty(block.Header);
```

Comment:
would be nice to add unit tests

Reply:
I don't think that `StartupBlockTreeFixer` can request orphaned blocks - it is starting from some canonical block and progressing up, so no chance for picking orphaned block. I'm not sure how other visitors works and generally we need this option in this specific one, so good idea with adding visitor option, I will do it